apps/vs-code-designer/src/assets/UnitTestTemplates/TestClassFile

```diff
@@ -1,77 +1,128 @@
-        public void ExecuteWorkflow_WithMockTrigger_ShouldSucceed()
+        /// <summary>
+        /// The full path to the connections JSON file.
+        /// </summary>
+        private readonly string connectionsPath;
+
+        /// <summary>
+        /// The full path to the parameters JSON file.
+        /// </summary>
+        private readonly string parametersPath;
+
+        /// <summary>
+        /// The full path to the local settings JSON file.
+        /// </summary>
+        private readonly string localSettingsPath;
+
+        /// <summary>
+        /// The mocked trigger outputs that are generated from a workflow run.
+        /// </summary>
+        public TriggerMock triggerMock;
+
+        /// <summary>
+        /// The mocked actions' outputs that are generated from a workflow run.
+        /// </summary>
+        public Dictionary<string, ActionMock> actionMocks;
+
+        /// <summary>
+        /// Initializes a new instance of the class named <see cref="<%= UnitTestName %>" />.
+        /// This constructor loads the necessary paths for the workflow, connections, parameters, and local settings JSON files.
+        /// The constructor also loads the mock data from the file named <%= UnitTestName %>_mock.json.
+        /// </summary>
+        public <%= UnitTestName %>()
         {
-            // Arrange – define mocks either from a static source or as a lambda function
-            var triggerMock = new OperationMock("When_a_HTTP_request_is_received", FlowStatus.Succeeded, new JObject()
-            {
-                { "body", new JObject() { { "isReleased", true }, { "isEnabled", false }, { "isFlag", false } } }
-            });
-
-            // Act 
-            var unitTestFlowRun = this.UnitTestExecutor.ExecuteWorkflow(this.WorkflowDefinition, this.Mocks, triggerMock).Result;
-
-            // Assert
-            Assert.IsNotNull(unitTestFlowRun);
-            Assert.AreEqual(FlowStatus.Succeeded, unitTestFlowRun.Properties.Status);
-            Assert.AreEqual("myvalue", unitTestFlowRun.Properties.Actions["Get_Rows_Receipt"].Outputs.Value<string>());
+            //Get the root path of the workspace
+            this.rootPath = Path.GetFullPath(Path.Combine(AppContext.BaseDirectory, @"..\..\..\..\.."));
+
+            // Build the full paths to the required JSON files.
+            this.workflowDefinitionPath = Path.Combine(rootPath, "<%= LogicAppName %>", "<%= WorkflowName %>", "workflow.json");
+            this.connectionsPath = Path.Combine(rootPath, "<%= LogicAppName %>", "connections.json");
+            this.parametersPath = Path.Combine(rootPath, "<%= LogicAppName %>", "parameters.json");
+            this.localSettingsPath = Path.Combine(rootPath, "<%= LogicAppName %>", "local.settings.json");
+
+            // Set path to the mock.json file.
+            var mockDataPath = Path.Combine(rootPath, "Tests", "<%= LogicAppName %>", "<%= WorkflowName %>", "<%= UnitTestName %>", "<%= UnitTestName %>-mock.json");
```

Comment:
path of mock data should always be same folder as this test class file right? So file name should suffice here right?

Reply:
fixed

Reply:
@nidhi-pathak Based on the PR feedback, it appears the folder could not be located at the specified path. To keep us on track, I’ll be reverting to the most recent working version for now.